### ISSUE TITLE
Gene Set Enrichment Analysis tool: does not merge genes from input list

### ISSUE THREAD

**Issue by chris-grove** [WormBase/website#6300]
I just noticed when running the Gene Set Enrichment Analysis tool that the tool returns the entire list of genes entered, regardless of whether they are duplicates or not. For example, if I use WOBr to collect the genes annotated to the "adult lethal" phenotype (or descendant terms), I can copy, verbatim, all of the gene names from WOBr, remove the list titles (so that I only have genes left) and I have a list of 178 genes, most of which are duplicated at least once as the non-redundant list of genes is really 89 genes. If I paste the original list of 178 gene names (with duplicates) into the GSEA tool and run it, I get the report that says that:

![screen shot 2018-03-29 at 11 28 49 am](https://user-images.githubusercontent.com/611382/38097833-6d9d2aa0-3344-11e8-9e15-13b119b281ee.png)

even though the list is actually of only 89 genes. I don't think this has impacted the results at all (as I seem to get the equivalent results when I run the analysis on the non-redundant list of 89 genes), but I would wonder if users would think this means that the tool did not remove redundancies and counted genes multiple times whenever there was a duplicate. I think the default behavior should be to remove duplicates and report the non-redundant list of genes analyzed.

**Comment by raymond91125** [comment]
I actually like it this way better for the sake of simplicity. A naive user would be alarmed if the numbers we give back (invalid+no data+tested) don't add up to what they put in. What I used to test is this set {eat-12, egl-19, eat-0, eat-8}.

**Comment by chris-grove** [comment]
@raymond91125 
I actually found it alarming that it DIDN'T remove the redundancies, because then I wonder if the system counts each duplicate in the calculation (which would obviously be bad). But, I'll defer the ultimate decision to you and others.

**Comment by raymond91125** [comment]
Would it be better if we remove the last list (included in analysis) completely? That is, whatever is not "not valid" or "no data" is included and no counts given.

**Comment by chris-grove** [comment]
Well, I thought that it was nice/useful to see those numbers, particularly if your list has a lot of "censored" genes due to lack of data, as that could really impact the conclusions.

**Comment by chris-grove** [comment]
Yes, I meant I like seeing both the "valid" and "invalid" list of genes and numbers as a sanity check. I guess it would be good to hear some other opinions on the issue. Maybe we can do as you suggest and remove the valid list (and number), but I do like seeing it, for the record.

**Comment by azurebrd** [comment]
@raymond91125 Did we fix this ?
